src/service/storage-impl.js

```diff
@@ -82,6 +82,16 @@ export class Storage {
     return this.getStore_().then((store) => store.get(name));
   }
 
+  /**
+   * Returns the promise that yields the timestamp of the property for the specified
+   * key.
+   * @param {string} name
+   * @return {!Promise<number|undefined>}
+   */
+  getTimestamp(name) {
```

Comment:
Instead of having individual caller handle the expiration of local storage. I think a more helpful method would be to provide a method like `getUnexpiredValue(name, duration)`. 
Not a blocker.